### PR TITLE
API - disable user functionality, quote statuses (pending, approved, …

### PR DESCRIPTION
**Pull request by kakucav**
API - disable user functionality, quote statuses (pending, approved, rejected)

### CODE REVIEW

**Comment by zlayabekrija** [comment]
[Live](https://ischoolprojects.github.io/ColorMagic/)

**Review by DejanMaksimovic — APPROVED** [review]

File: src/admin/admin-quote/admin-quote.service.ts [hunk 1]
```diff
@@ -69,20 +76,25 @@ export class AdminQuoteService {
     const quote: Quote = new Quote(adminCreateQuoteDto);
     quote.slug = await this.helperService.generateSlug(quote.title);
```
Comment (zlayabekrija) [inline src/admin/admin-quote/admin-quote.service.ts:77]: **Should:** this should be done, with after create hook.

File: src/auth/auth.service.ts [hunk 2]
```diff
@@ -44,6 +48,8 @@ export class AuthService {
 
     await this.checkPassword(password, user);
 
+    this.checkIfUserIsEnabled(user.isEnabled);
```
Comment (zlayabekrija) [inline src/auth/auth.service.ts:51]: **Should:** by the time we reach password check, we should already know if the user is disabled.

File: src/admin/admin-quote/admin-quote.service.ts [hunk 3]
```diff
@@ -69,20 +76,25 @@ export class AdminQuoteService {
...
 
-    return await this.quoteRepository.save(quote);
+    return await this.quoteRepository.save({
+      ...quote,
+      status: QuoteStatus.APPROVED,
```
Comment (zlayabekrija) [inline src/admin/admin-quote/admin-quote.service.ts:81]: **Should:** if the quote is approved by default, why shouldn't we set it on entity/table level.

**Review by zlayabekrija — COMMENTED** [review]
Great work, please address these minor issues.

**Review by zlayabekrija — APPROVED** [review]
Great effort :100:

**Pull request merged**